api/orchestrator/orchestrator.proto

```diff
@@ -423,31 +460,70 @@ message CloudService {
   string name = 2;
 
   string description = 3;
+}
+
+message Catalog {
+  string id = 1;
+
+  string name = 2;
 
-  message Requirements { repeated string requirement_ids = 1; }
+  string description = 3;
 
-  Requirements requirements = 4;
+  repeated Control controls = 4;
 }
 
-// Requirement represents a certain requirement that needs to be fulfilled. It
-// could be a control in a certification catalog.
-message Requirement {
+// Control represents a certain Control that needs to be fulfilled. It
+// could be a Control in a certification catalog.
+message Control {
   string id = 1;
 
   string name = 2;
 
   string description = 3;
 
+  repeated Control controls = 4
+      [ (tagger.tags) = "gorm:\"serializer:json\"" ];
```

Comment:
TODO: Discuss internally how to do that.

Some thoughts:
- I imagine with json, we store too much directly here? When retrieving a control, it will act like preloading everything. Maybe we even want it that way?
- Using a field `string control_id` will maybe implicitly create the desired nested controls (leaving it empty means that is the root control)
- many2many tag or similar to do the nesting? Dunno if that is working here

Reply:
This should be a simple belongs to I think https://gorm.io/docs/belongs_to.html

Reply:
I did it this way now (= option 2 w/ `string control_id`). But now we have the following (see also the failing test):
Imagine having a catalog with one control (Cont1234). And this control again has a sub control (Cont1234.1). Which again has a sub control (Cont1234.1.1).
When we now fetch the catalog, the list of controls includes all three controls - although we should only get the one main control. That's the case since they all refer to the same catalog with the corresponding id.

Some options:
- For sub controls we leave the `catalog_id` field empty
- We define a message sub_control that has no field `catalog_id` and use them for sub controls
- (We leave it like that since we get all controls this way. But then the client has to clean up the control mess)

Nevertheless, I think we need some API point for fetching controls, e.g. list/get control. Otherwise we have no chance to get any sub controls due to our restriction to only preload one level deep.

Reply:
> I did it this way now (= option 2 w/ `string control_id`). But now we have the following (see also the failing test): Imagine having a catalog with one control (Cont1234). And this control again has a sub control (Cont1234.1). Which again has a sub control (Cont1234.1.1). When we now fetch the catalog, the list of controls includes all three controls - although we should only get the one main control. That's the case since they all refer to the same catalog with the corresponding id.
> 
> Some options:
> 
> * For sub controls we leave the `catalog_id` field empty
> * We define a message sub_control that has no field `catalog_id` and use them for sub controls
> * (We leave it like that since we get all controls this way. But then the client has to clean up the control mess)
> 
> Nevertheless, I think we need some API point for fetching controls, e.g. list/get control. Otherwise we have no chance to get any sub controls due to our restriction to only preload one level deep.

Only the top most control should have the `catalog_id` field set. In fact, you will probably not need to set the field itself, gorm will do it for you.

Reply:
Additionally, we can also think about changing the `control_id` field to `parent_id` or `parent_control_id` to make it more readable that this only applies to sub-controls.